### PR TITLE
add handling for missing filetype

### PR DESCRIPTION
**Pull request by kkoehn**
Adds handling of missing filetype (mainly during proforma import)

### CODE REVIEW

File: app/models/exercise_file.rb [hunk 1]
```diff
@@ -55,4 +55,10 @@ def parse_text_data
     self.content = Paperclip.io_adapters.for(attachment.instance.attachment).read
     self.attachment = nil
   end
+
+  def file_type_by_extension(extension)
+    FileType.find_or_create_by(file_extension: extension) do |filetype|
```
Comment (MrSerth) [inline app/models/exercise_file.rb:60]: Should we use `file_type` here as well? Similar to CodeOcean? And should be similar to `spec/factories/file_type.rb`

Comment (kkoehn) [reply app/models/exercise_file.rb:60]: I used `filetype` to avoid a name collision - the method is in exercise_file which has a `belongs_to :file_type`

Comment (kkoehn) [reply app/models/exercise_file.rb:60]: In the factory  I tried to stay coherent

Comment (MrSerth) [reply app/models/exercise_file.rb:60]: Ah, right. Thanks - I didn't see that in the collapsed GitHub view.
I agree with you; we should keep it coherent.

**Pull request merged**